### PR TITLE
[TRTLLM-16326][test] Fail VisualGen tests on missing checkpoints/deps

### PR DESCRIPTION
**Pull request by zhenhuaw-me**
## Description
Missing model checkpoints — and other resources CI is expected to provide — in the VisualGen tests were silently `pytest.skip`-ped, so a test that could not actually run still reported green and its breakage dropped off CI's radar. This makes those cases fail loudly. Checkpoint resolution is unified on a single `get_checkpoint()` helper in `tests/test_common/llm_data.py` that resolves a model subdir under `LLM_MODELS_ROOT` and raises `FileNotFoundError` when it is absent, replacing ~22 duplicated local `_llm_models_root`/`_checkpoint` helpers and every per-model path env override (`DIFFUSION_MODEL_PATH_*`, `FLUX*_MODEL_PATH`, `LTX2_*`, `QWEN_IMAGE_CKPT`, `TRTLLM_CACHE_DIT_*`) — local-dev knobs that don't belong in committed CI tests. The unit suite, the integration benchmark, and the integration example suite (flux/ltx2/wan/qwen/hunyuan/multi_gpu/cosmos3/glm) all resolve through it, and the shared example-suite `_skip_if_missing` helper is replaced by a fail-loud `_require_exists`. Resolution is lazy (inside fixtures/tests, never at import) — including LTX-2, whose module-level constants previously ran `llm_models_root(check=True)` at collection — so the coupled CUDA / SM-arch / GPU-count skips still apply first and a missing checkpoint never becomes a collection-time error.
Beyond checkpoints, other silent skips that hid real breakage now fail loudly:
- `cache-dit` is a hard requirement (requirements.txt), so its `find_spec` skips in the Wan 2.2 T2V/I2V/TI2V pipelines and `test_cache_dit.py` are removed.
- The `multi_gpu` parallel harnesses (wan / flux2 / wan_vsa) no longer catch `(ImportError, ValueError, NotImplementedError)` around model construction and skip, and their first-party `try/except ImportError: MODULES_AVAILABLE` guards are gone.
- Cosmos3 attn2d and the wan/flux2 transformer-parallel harnesses assert FA4 / `flash_attn_combine` availability instead of skipping (these files are Blackwell-only, so a missing kernel is a build failure); missing-`torch.ops` and checkpoint-content-mismatch skips also fail.
- A missing golden-media asset (`_require_exists`, on committed Git-LFS artifacts) now fails rather than green-skipping a comparison that validates nothing, and the Qwen examples assert on an incomplete checkpoint (dir present but missing `model_index.json`) instead of skipping.
Genuinely-environmental gates stay skips: CUDA availability, SM/arch capability, GPU count, per-test config/workload preconditions, and optional deps (CuTe-DSL — including the wan VSA CuTe-DSL gate — and ffmpeg). One known gap is deliberately left as a skip and tracked as a follow-up in NVBug 6762388 rather than red-walling the shared B200 stage: the Cosmos3 Edge `DIFFUSERS_MAIN_PATH` parity test, which needs an external diffusers-main source checkout CI never stages and is unrunnable while diffusers is pinned to 0.39. Also adds `AGENTS.md`/`CLAUDE.md` under the VisualGen unit and integration test dirs codifying this policy, a `tests/`-level `AGENTS.md`/`CLAUDE.md` requiring every new test to be in a CI test list, a focused unit test for `get_checkpoint`, and l0_cpu entries for `visual_gen/test_utils.py` and the new `utils/test_llm_data.py`. No `tensorrt_llm` source change.

## Follow-up NVBugs
Filed separately (out of scope for this PR). The first tracks the one skip left in place; the rest track a related audit that found 12 VisualGen test files in no CI test list (never collected), grouped per model:
- NVBug 6762388 — Cosmos3 Edge `DIFFUSERS_MAIN_PATH` parity test not validated in CI (diffusers 0.39 pin vs Edge classes needing >=0.40).
- NVBug 6762589 — LTX-2 test not in any CI list (Yiyun Lu).
- NVBug 6762590 — Qwen-Image tests (6) not in any CI list (Yibin Li).
- NVBug 6762593 — Wan tests (2) not in any CI list (Olivia Stoner).
- NVBug 6762594 — VisualGen attention-backend tests (2) not in any CI list (Ruqing Xu).
- NVBug 6762595 — Executor / shared-tensor IPC test not in any CI list (Yiyun Lu).

## Test Coverage
Test-infra change, verified on a GB200 node against a precompiled 1.3.0rc26 wheel. `tests/unittest/utils/test_llm_data.py` passes 2/2 (successful resolution + missing-checkpoint `FileNotFoundError`); the edited unit files collect 153 tests with no new collection errors; `visual_gen/test_utils.py` passes 11/11 including the Gloo 4-rank distributed case; the formerly `@requires_cache_dit` CPU classes in `test_cache_dit.py` pass 9/9 without the dependency skip; and the integration benchmark plus the full example suite (including the golden/incomplete-checkpoint and cosmos3/glm conversions) collect 68 tests with no errors.

## PR Checklist
Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.
- [x] Please check this after reviewing the above items as appropriate for this PR.

## GitHub Bot Help
To see a list of available CI bot commands, please comment `/bot help`.

## Dev Engineer Review

- Centralizes checkpoint resolution in `get_checkpoint()` under `LLM_MODELS_ROOT`.
- Missing checkpoints and required resources now raise failures instead of skips.
- Removes duplicated path helpers and per-model environment overrides.
- Required dependencies, operators, kernels, model construction, checkpoint content, and golden media now fail tests.
- Preserves valid platform, CUDA, GPU-count, workload, optional-dependency, and external-source skips.
- Adds test-policy guidance and CPU test-list entries.
- No `tensorrt_llm` source changes are included.
- Main risk: environments with missing checkpoints or dependencies will now fail during collection or fixture setup.

## QA Engineer Review

- Updates VisualGen unit, multi-GPU, benchmark, and integration tests.
- Adds `get_checkpoint()` success and missing-checkpoint tests.
- Adds utility tests to `tests/integration/test_lists/test-db/l0_cpu.yml`.
- Changes missing-resource, dependency, operator, kernel, model-construction, checkpoint-content, and golden-media handling from skips to failures.
- Retains CUDA, hardware, GPU-count, workload, optional-dependency, and external-source gates.
- Coverage verdict: **needs follow-up**. CPU utility coverage is listed, but CI or manual-QA coverage for all modified GPU and integration tests is not confirmed.

## Per-File QA Perspective

- `tests/test_common/llm_data.py`: Adds centralized checkpoint resolution and `FileNotFoundError` handling. Verify root validation, subdirectory resolution, and lazy evaluation.
- `tests/unittest/utils/llm_data.py`: Re-exports `get_checkpoint`. Verify existing exports remain unchanged.
- `tests/unittest/utils/test_llm_data.py`: Covers valid and missing checkpoint resolution. Listed in `l0_cpu.yml`.
- `tests/integration/test_lists/test-db/l0_cpu.yml`: Adds CPU entries for utility tests. Verify paths remain valid.
- `tests/unittest/_torch/visual_gen/*.py`: Replaces local checkpoint and skip logic with centralized resolution. Verify valid environment gates and required-resource failures.
- `tests/unittest/_torch/visual_gen/multi_gpu/*.py`: Removes required-module skips and makes kernel and model-construction failures visible. Verify distributed setup and GPU-count gates.
- `tests/integration/defs/examples/visual_gen/*.py`: Requires checkpoints and test assets for VisualGen example flows. Verify missing resources fail and platform gates still skip.
- `tests/integration/defs/examples/visual_gen/visual_gen_test_utils.py`: Makes missing golden media, golden files, and Wan checkpoints fail. Verify all callers handle this behavior.
- `tests/*/AGENTS.md` and `tests/*/CLAUDE.md`: Adds test-policy and CI-list guidance. Verify the guidance identifies the correct test lists.

### CODE REVIEW

File: tests/integration/defs/visual_gen/test_visual_gen_benchmark.py [hunk 1]
```diff
@@ -43,13 +43,12 @@
 
 
 def _wan_t2v_path() -> Path:
```
Comment (zhenhuaw-me) [inline tests/integration/defs/visual_gen/test_visual_gen_benchmark.py]: why not use get_checkpoint?

File: tests/test_common/llm_data.py [hunk 2]
```diff
@@ -86,6 +86,17 @@ def llm_models_root(check: bool = False) -> Optional[Path]:
     return root if root.exists() else None
 
 
+def get_checkpoint(model_subdir: str) -> str:
+    """Resolve a checkpoint under LLM_MODELS_ROOT, or fail loudly if missing."""
```
Comment (zhenhuaw-me) [inline tests/test_common/llm_data.py]: Could we apply this fail-loud contract to the integration example suite as well? `tests/integration/defs/examples/visual_gen/visual_gen_test_utils.py::_skip_if_missing()` still skips missing checkpoints/golden assets for roughly 40 listed tests, the Wan/Qwen examples have direct checkpoint skips, and the Cosmos3 Edge parity test still skips when `DIFFUSERS_MAIN_PATH` is unset.

File: tests/unittest/_torch/visual_gen/multi_gpu/test_cosmos3_transformer_parallel.py [hunk 3]
```diff
@@ -55,7 +47,7 @@
         _flash_attn_combine as _fa_combine,
     )
 
-    _ATTN2D_AVAILABLE = MODULES_AVAILABLE and _fa4_fwd is not None and _fa_combine is not None
+    _ATTN2D_AVAILABLE = _fa4_fwd is not None and _fa_combine is not None
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/multi_gpu/test_cosmos3_transformer_parallel.py]: Could we remove this availability-based skip for the B200-listed attn2d tests? Missing FA4/`flash_attn_combine` JIT kernels on their intended runner is a build/dependency failure, while the GPU-count checks already cover unsupported hardware.

File: tests/unittest/_torch/visual_gen/multi_gpu/test_wan_transformer_parallel.py [hunk 4]
```diff
@@ -35,19 +35,14 @@
...
-        AttentionConfig,
-        DiffusionModelConfig,
-        TorchCompileConfig,
-    )
-    from tensorrt_llm._torch.visual_gen.mapping import VisualGenMapping
-
-    from .tp_shard_utils import copy_tp_parameter
+from tensorrt_llm._torch.visual_gen.config import (
+    AttentionConfig,
+    DiffusionModelConfig,
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/multi_gpu/test_wan_transformer_parallel.py]: Could we also let model-construction errors fail instead of catching `(ImportError, ValueError, NotImplementedError)` and skipping below? The same pattern remains in `test_flux2_transformer_parallel.py` and `test_wan_vsa_ulysses.py`, so real backend/config regressions can still report green.

File: tests/unittest/_torch/visual_gen/test_wan22_t2v_pipeline.py [hunk 5]
```diff
@@ -463,10 +430,8 @@ class TestWan22T2VCombinedOptimizations:
 
     @pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available")
     def test_fp8_cache_dit_trtllm(self):
-        if not os.path.exists(WAN22_A14B_PATH):
-            pytest.skip(f"Checkpoint not found: {WAN22_A14B_PATH}")
         args = VisualGenArgs(
-            model=WAN22_A14B_PATH,
+            model=get_checkpoint(WAN22_A14B_SUBDIR),
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/test_wan22_t2v_pipeline.py]: Could we remove the `find_spec("cache_dit")` skip on this class? `cache-dit` is already in `requirements.txt` and these tests are in the B200 test list; the same dependency skip remains in the I2V, TI2V, and `test_cache_dit.py` coverage.

File: tests/unittest/_torch/visual_gen/test_ltx2_pipeline.py [hunk 6]
```diff
@@ -51,18 +51,15 @@
 
 
 _LTX2_BASE = os.path.join(str(llm_models_root(check=True)), "LTX-2")
-_GEMMA3_DEFAULT = os.path.join(str(llm_models_root(check=True)), "gemma", "gemma-3-12b-it")
 
+_LTX2_BF16_SUBDIR = "LTX-2/ltx-2-19b-dev.safetensors"
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/test_ltx2_pipeline.py]: Could we remove `_LTX2_BASE` and the remaining module-level paths above? They still call `llm_models_root(check=True)` during import, so collection can fail before CUDA/platform markers or the new lazy fixtures run; the tests should use the fixture return values/`get_checkpoint()` lazily.

File: tests/test_common/llm_data.py [hunk 7]
```diff
@@ -86,6 +86,17 @@ def llm_models_root(check: bool = False) -> Optional[Path]:
...
+    root = llm_models_root(check=True)
+    path = root / model_subdir
+    if not path.exists():
```
Comment (zhenhuaw-me) [inline tests/test_common/llm_data.py]: Could we add a focused unit test for successful resolution and the missing-checkpoint error? This helper is the central fail-loud contract of the PR, but the failure path is only described as manually exercised.

File: tests/unittest/_torch/visual_gen/test_flux_pipeline.py [hunk 8]
```diff
@@ -137,23 +126,13 @@ def _find_first_quantizable_linear(transformer):
 
 @pytest.fixture
 def flux1_checkpoint_exists():
-    """Check if FLUX.1 checkpoint is available locally."""
-    if not FLUX1_CHECKPOINT_PATH or not os.path.exists(FLUX1_CHECKPOINT_PATH):
-        pytest.skip(
-            f"FLUX.1 checkpoint not found at {FLUX1_CHECKPOINT_PATH}. "
-            "Set FLUX1_MODEL_PATH or stage checkpoint under LLM_MODELS_ROOT."
-        )
+    """FLUX.1 checkpoint gate; the path is resolved (and staging enforced) at import."""
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/test_flux_pipeline.py]: Could we remove these no-op fixtures and their unused parameters? They no longer resolve or gate a checkpoint, and the docstrings incorrectly say resolution happens at import; non-blocking cleanup.

File: tests/unittest/_torch/visual_gen/test_utils.py [hunk 9]
```diff
@@ -23,13 +23,8 @@
...
 import torch.multiprocessing as mp
 
-try:
-    from tensorrt_llm._torch.visual_gen.utils import SequenceSharder
-    from tensorrt_llm._utils import get_free_port
-
-    MODULES_AVAILABLE = True
-except ImportError:
-    MODULES_AVAILABLE = False
+from tensorrt_llm._torch.visual_gen.utils import SequenceSharder
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/test_utils.py]: Can we add this file to the CPU test list? All tests are marked `cpu_only` and the distributed case uses Gloo, so removing the import skip otherwise adds no CI coverage.

File: tests/AGENTS.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# AGENTS.md — tests
+
+Scope: this directory and everything under it. Supplements the repo-root `AGENTS.md`.
+
+## Every test must be in a CI test list
+
+A new test file — or a new test that needs its own entry — MUST be added to the
```
Comment (chzblych) [inline tests/AGENTS.md]: I wonder whether every new test must be added to the CI test lists. For example, some tests may be intended to run only in the developer’s local env or as part of QA testing. My understanding is that we don’t aim to run every test in CI.

Perhaps we could phrase it this way:

_If a newly added test needs to be protected by CI, it must be included in at least one CI test list. Tests that are not included in any CI test list are considered intended for local developer testing or QA testing rather than CI._

**Review by chzblych — APPROVED** [review]

File: tests/AGENTS.md (hunk 10 above)
Comment (BowenFu) [reply tests/AGENTS.md]: +1

**Review by BowenFu — APPROVED** [review]

File: tests/AGENTS.md (hunk 10 above)
Comment (zhenhuaw-me) [reply tests/AGENTS.md]: Good point — reworded. The reminder is now that confirm with the author whether it should run in CI and which platform list should carry it when add a test.

File: tests/integration/test_lists/waives.txt [hunk 11]
```diff
@@ -79,6 +79,8 @@ examples/visual_gen/test_visual_gen_ltx2.py::test_ltx2_feature_accuracy_against_
 examples/visual_gen/test_visual_gen_ltx2.py::test_ltx2_feature_accuracy_against_golden[nvfp4] SKIP (https://nvbugs/6572800)
 examples/visual_gen/test_visual_gen_ltx2.py::test_ltx2_lpips_against_golden SKIP (https://nvbugs/6759067)
 examples/visual_gen/test_visual_gen_qwen_image.py::test_qwen_image_edit_example SKIP (https://nvbugs/6726626)
+examples/visual_gen/test_visual_gen_qwen_image.py::test_qwen_image_layered_example SKIP (https://nvbugs/6771010)
```
Comment (BowenFu) [inline tests/integration/test_lists/waives.txt:82]: Could you add the current NVBug evidence for 6771010—an affected CI run/head and the exact missing checkpoint—to the PR description? These two new waivers suppress the layered Qwen-Image checks, but the current follow-up list does not mention 6771010, so the intended skip and scope cannot be verified from the PR. This evidence is required for the PR rather than a follow-up.

Comment (zhenhuaw-me) [reply tests/integration/test_lists/waives.txt:82]: Added to the bug.

**Review by BowenFu — APPROVED** [review]

**Review by chang-l — APPROVED** [review]
